torch/csrc/autograd/engine.cpp

```diff
@@ -332,6 +336,19 @@ auto Engine::evaluate_function(FunctionTask& task) -> void {
 
   int num_outputs = outputs.size();
   if (num_outputs == 0) return; // Don't even acquire the mutex
+
+  if (AnomalyMode::is_enabled()) {
+    AutoGradMode grad_mode(false);
+    for (int i = 0; i < num_outputs; ++i) {
+      auto& output = outputs[i];
+      if (output.ne(output).any().toCByte()) {
+        std::stringstream ss;
+        ss << "Function '" << fn.name() << "' returned nan values in it's " << i << "th output.";
```

Comment:
nit: its
nit: do we have a function for getting 1st, 2nd, 3rd, correct?

Reply:
I can't find one.